### PR TITLE
Bug 1189641 - fix: move security related settings reading to state manager

### PR DESCRIPTION
**Pull request by snowmantw**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=73b637604fac857b4b0c4f6b0877cfa155fe0ec4

File: apps/system/lockscreen/js/lockscreen.js [hunk 1]
```diff
@@ -120,11 +101,6 @@
     */
     _slideReachEnd: false,
 
-    /*
-    * Current passcode entered by the user
-    */
-    passCodeEntered: '',
```
Comment (snowmantw) [inline apps/system/lockscreen/js/lockscreen.js]: remove legacy code: it's now handled by input pad

File: apps/system/lockscreen/js/lockscreen.js [hunk 2]
```diff
@@ -252,29 +216,12 @@
...
-      case 'lockscreenslide-near-right':
-        break;
       case 'lockscreenslide-unlocking-start':
         this._notifyUnlockingStart();
         break;
       case 'lockscreenslide-unlocking-stop':
         this._notifyUnlockingStop();
         break;
-      case 'lockscreenslide-activate-left':
-      case 'holdcamera':
```
Comment (snowmantw) [inline apps/system/lockscreen/js/lockscreen.js]: moved to the state manager

File: apps/system/lockscreen/js/lockscreen.js [hunk 3]
```diff
@@ -380,14 +322,6 @@
       window.addEventListener('iac-mediacomms', this);
       window.addEventListener('appterminated', this);
 
-      // Listen to event to start the Camera app
-      window.addEventListener('holdcamera', this);
-
-      window.SettingsListener.observe('lockscreen.enabled', true,
```
Comment (snowmantw) [inline apps/system/lockscreen/js/lockscreen.js]: Enabled or not now is in LockScreenWindowManager

File: apps/system/lockscreen/js/lockscreen.js [hunk 4]
```diff
@@ -433,13 +353,9 @@
       // keep track of its value.
       navigator.mozSettings.addObserver('lockscreen.lock-immediately',
         (function(event) {
-        if (event.settingValue === true) {
-          this.lockIfEnabled(true);
-        }
+          this.lock(true);
```
Comment (snowmantw) [inline apps/system/lockscreen/js/lockscreen.js]: In this patch I don't want to bother FMD. Move it later.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=00f05e3a371ed6d54a469e00a86b89205965941c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=afcf7dc32fbdfa7c932379be88840adda38783ec

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ee444c68f6114ba311d6b1a5f7d7f8c090ae0451

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ee444c68f6114ba311d6b1a5f7d7f8c090ae0451

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a0f15d0e90899128deb18e96f1ec38d9ac93218d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3610f82d9497d958a01a7af87db15c0b98c7074a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ecbb272e953904d5bc3ec1c3f8446b4aa4f19e3d

File: apps/system/lockscreen/js/lockscreen.js [hunk 5]
```diff
@@ -527,76 +443,12 @@
...
     this.message.textContent = val;
     this.message.hidden = (val === '');
   };
 
-  /**
-   * Light the camera and unlocking icons when user touch on our LockScreen.
-   *
-   * @this {LockScreen}
-   */
-  LockScreen.prototype._lightIcons =
```
Comment (timdream) [inline apps/system/lockscreen/js/lockscreen.js]: Note to myself: dead code.

File: apps/system/lockscreen/js/lockscreen_state_manager.js [hunk 6]
```diff
@@ -201,7 +216,7 @@
 
     this.registerRule({
       passcodeEnabled: true,
-      passcodeTimeout: false,
+      passcodeTimeoutExpired: false,
```
Comment (timdream) [inline apps/system/lockscreen/js/lockscreen_state_manager.js]: Nice naming change.

File: apps/system/lockscreen/js/lockscreen_state_manager.js [hunk 7]
```diff
@@ -654,35 +691,24 @@
...
-      // the manager keeps. This is due to some request, like pressing homekey,
-      // is a one-time event, and will not update the inner states table.
-      // So if an event is changing the inner states table, we need to
-      // toggle the value here, not in the resolver of postponed request.
-      this.lockScreenStates.passcodeEnabled = val;
-      resolve(val);
-    } else {
-      this.lockScreenStates.passcodeEnabled = val;
-    }
+  LockScreenStateManager.prototype.onSettingUpdated =
```
Comment (timdream) [inline apps/system/lockscreen/js/lockscreen_state_manager.js]: `s/onSettingUpdated/getSettingsObserverCallback/`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2eb85baa8f09eec3e02abc34263df6761fc0e538

**Pull request merged**